modules/mobile/app/controllers/mobile/v0/payment_history_controller.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require_dependency 'mobile/application_controller'
+require 'adapters/payment_shared_adapter'
+
+module Mobile
+  module V0
+    class PaymentHistoryController < ApplicationController
+      def index
+        start_date = params[:startDate] || (DateTime.now.utc.beginning_of_day - 1.year).iso8601
+        end_date = params[:endDate] || DateTime.now.utc.beginning_of_day.iso8601
+
+        validated_params = Mobile::V0::Contracts::GetPaginatedList.new.call(
+          start_date: start_date,
+          end_date: end_date,
+          page_number: params.dig(:page, :number),
+          page_size: params.dig(:page, :size),
+          use_cache: false,
+          reverse_sort: false
+        )
+
+        person = BGS::PeopleService.new(current_user).find_person_by_participant_id
+        response = BGS::PaymentService.new(current_user).payment_history(person)
+        payments = adapt_payments(response.dig(:payments, :payment))
+        list, meta = paginate(payments, validated_params)
+        render json: Mobile::V0::PaymentHistorySerializer.new(list, meta)
+      end
+
+      private
+
+      def adapt_payments(payments)
+        payments.map do |payment|
+          if payment.dig(:return_payment, :check_trace_number).blank?
+            Mobile::V0::PaymentHistory.new(
+              id: payment.dig(:payment_record_identifier, :payment_id),
+              amount: ActiveSupport::NumberHelper.number_to_currency(payment[:payment_amount]),
+              date: payment[:payment_date],
+              payment_method: ::Adapters::PaymentSharedAdapter.new.get_payment_method(payment),
+              bank: payment.dig(:address_eft, :bank_name),
+              account: ::Adapters::PaymentSharedAdapter.new.mask_account_number(payment[:address_eft])
+            )
+          end
+        end
+      end
+
+      def paginate(payments, validated_params)
+        # payments = payments.filter do |payment|
```

Comment:
What's up with this?

Reply:
That is for the pagination functionality (which I'll be working on this sprint)